root/programs/C#/Frameworks/Infrastructure/Business/Presentation/MyBaseApiControllerAsync.cs

```diff
@@ -0,0 +1,385 @@
+﻿//**********************************************************************************
+//* Copyright (C) 2007,2016 Hitachi Solutions,Ltd.
+//**********************************************************************************
+
+#region Apache License
+//  
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. 
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#endregion
+
+//**********************************************************************************
+//* クラス名        ：MyBaseApiControllerAsync (ActionFilterAttribute)
+//* クラス日本語名  ：非同期 ASP.NET WebAPI用 ベーククラス２（テンプレート）
+//*
+//* 作成者          ：生技 西野
+//* 更新履歴        ：
+//* 
+//*  日時        更新者            内容
+//*  ----------  ----------------  -------------------------------------------------
+//*  2017/08/11  西野 大介         新規作成
+//**********************************************************************************
+
+#pragma warning disable 1998
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Security.Claims;
+
+using System.Web;
+using System.Web.Http.Controllers;
+using System.Web.Http.Filters;
+using System.Net.Http;
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+using Touryo.Infrastructure.Framework.Authentication;
+using Touryo.Infrastructure.Framework.Exceptions;
+using Touryo.Infrastructure.Public.Log;
+using Touryo.Infrastructure.Public.Util;
+
+namespace Touryo.Infrastructure.Business.Presentation
+{
+    /// <summary>非同期 ASP.NET WebAPI用 ベーククラス２</summary>
+    /// <remarks>（ActionFilterAttributeとして）自由に利用できる。</remarks>
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, Inherited = true, AllowMultiple = true)]
+    public class MyBaseApiControllerAsync : ActionFilterAttribute, IAuthenticationFilter, IActionFilter, IExceptionFilter
```

Comment:
@daisukenishino 

The class name should be a semantic name (what does this class do?). So, I think that the class name should not contain the word *Async*.
The name of asynchronous method should contain the word *Async*.